src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherV3Test.java

```diff
@@ -0,0 +1,276 @@
+package org.broadinstitute.dsde.consent.ontology.datause;
+
+import org.broadinstitute.dsde.consent.ontology.model.DataUseV3;
+import org.broadinstitute.dsde.consent.ontology.model.DataUseBuilderV3;
+import org.broadinstitute.dsde.consent.ontology.model.TermParent;
+import org.broadinstitute.dsde.consent.ontology.model.TermResource;
+import org.broadinstitute.dsde.consent.ontology.service.AutocompleteService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.broadinstitute.dsde.consent.ontology.datause.MatchResultType.Abstain;
+import static org.broadinstitute.dsde.consent.ontology.datause.MatchResultType.Approve;
+import static org.broadinstitute.dsde.consent.ontology.datause.MatchResultType.Deny;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class DataUseMatcherV3Test {
+
+  private static final String cancerNode = "http://purl.obolibrary.org/obo/DOID_162";
+  private static final String intestinalCancerNode = "http://purl.obolibrary.org/obo/DOID_10155";
+
+  @Mock
+  private AutocompleteService autocompleteService;
+
+  @BeforeEach
+  public void setUp() {
+    openMocks(this);
+    try {
+      when(autocompleteService.lookupById(any())).thenReturn(Collections.emptyList());
+    } catch (Exception e) {
+      //
+    }
+  }
+
+  @Test
+  public void testDiseaseMatching_positive() throws Exception {
+    DataUseV3 dataset = new DataUseBuilderV3()
+        .setDiseaseRestrictions(Collections.singletonList(cancerNode))
+        .build();
+    DataUseV3 purpose = new DataUseBuilderV3()
+        .setDiseaseRestrictions(Collections.singletonList(intestinalCancerNode))
+        .build();
+    // Build a mock response of term parents based on what is returned when searching on DOID_10155
+    List<TermResource> termResources = new ArrayList<>();
+    TermResource resource = new TermResource();
+    TermParent parent1 = new TermParent();
+    parent1.setId("http://purl.obolibrary.org/obo/DOID_3119");
+    TermParent parent2 = new TermParent();
+    parent2.setId("http://purl.obolibrary.org/obo/DOID_0050686");
+    TermParent parent3 = new TermParent();
+    parent3.setId("http://purl.obolibrary.org/obo/DOID_162");
+    TermParent parent4 = new TermParent();
+    parent4.setId("http://purl.obolibrary.org/obo/DOID_14566");
+    TermParent parent5 = new TermParent();
+    parent5.setId("http://purl.obolibrary.org/obo/DOID_4");
```

Comment:
I wonder if we could make these constants at the top of the class to make it easier to follow in the tests.